### PR TITLE
HSEARCH-4166 + HSEARCH-4177 Fix, document and test named predicates more extensively

### PR DESCRIPTION
**Pull request by yrodiere**
* [HSEARCH-4177](https://hibernate.atlassian.net/browse/HSEARCH-4177): Improve NamedPredicate API
* [HSEARCH-4166](https://hibernate.atlassian.net/browse/HSEARCH-4166): Document and test named predicates more extensively

### CODE REVIEW

File: documentation/src/main/asciidoc/reference/mapper-orm-bridge-named-predicate.asciidoc [hunk 1]
```diff
@@ -0,0 +1,69 @@
+[[mapper-orm-bridge-named-predicate]]
+= Defining named predicates
+
+When implementing a <<mapper-orm-bridge-propertybridge,`PropertyBinder`>>
+or <<mapper-orm-bridge-typebridge,`TypeBinder`>>,
+it is possible to assign "named predicates"
+to index schema elements (either the index root or an <<mapper-orm-bridge-index-field-dsl-object,object field>>).
+
+These named predicates will then be usable as "black boxes" <<search-dsl-predicate-named,through the Search DSL>>,
```
Comment (fax4ever) [inline documentation/src/main/asciidoc/reference/mapper-orm-bridge-named-predicate.asciidoc:9]: black boxes? Maybe we can use a different term for it

File: documentation/src/main/asciidoc/reference/mapper-orm-bridge-named-predicate.asciidoc [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+.Declaring a named predicate
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/mapper/orm/bridge/namedpredicate/SkuIdentifierBinder.java[tags=binder]
+----
+<1> The binder declares its dependencies on the bridged element, as <<mapper-orm-bridge-propertybridge,usual>>.
+<2> The binder declares the index fields and their type, as <<mapper-orm-bridge-propertybridge,usual>>.
+<3> The binder defines a bridge to populate the fields from the bridged element, as <<mapper-orm-bridge-propertybridge,usual>>.
```
Comment (fax4ever) [inline documentation/src/main/asciidoc/reference/mapper-orm-bridge-named-predicate.asciidoc:23]: If there isn't anything new with that, maybe it is better to remove these points.

**Review by fax4ever — COMMENTED** [review]
Just 2 comments on this very extensive contrubution!

**Comment by yrodiere** [comment]
Addressed your comments and merged. Thanks.

**Pull request merged**